flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/assigner/MongoSplitAssigner.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.mongodb.source.enumerator.assigner;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.connector.mongodb.source.enumerator.MongoSourceEnumState;
+import org.apache.flink.connector.mongodb.source.split.MongoSourceSplit;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Optional;
+
+/** The split assigner for {@link MongoSourceSplit}. */
+@Internal
+public interface MongoSplitAssigner extends Serializable {
+
+    /**
+     * Called to open the assigner to acquire any resources, like threads or network connections.
+     */
+    void open();
+
+    /**
+     * Called to close the assigner, in case it holds on to any resources, like threads or network
+     * connections.
+     */
+    void close() throws IOException;
+
+    /** Gets the next split. */
+    Optional<MongoSourceSplit> getNext();
```

Comment:
When all splits have been assigned, but the enumerator has not signaled no more splits yet.
When the enumerator receives an empty split, it will check whether it should notify the reader to close.

`MongoSourceEnumerator#assignSplits`
```java
Optional<MongoSourceSplit> split = splitAssigner.getNext();
if (split.isPresent()) {
    final MongoSourceSplit mongoSplit = split.get();
    context.assignSplit(mongoSplit, nextAwaiting);
    awaitingReader.remove();
    LOG.info("Assign split {} to subtask {}", mongoSplit, nextAwaiting);
    break;
} else if (splitAssigner.noMoreSplits() && boundedness == Boundedness.BOUNDED) {
    LOG.info("All splits have been assigned");
    context.registeredReaders().keySet().forEach(context::signalNoMoreSplits);
    break;
} else {
    // there is no available splits by now, skip assigning
    break;
}
```

Reply:
> Document that. Specifically it's relationship to `noMoreSplits()`.

Add document : `Gets the next split to assign to MongoSourceSplitReader when MongoSourceEnumerator receives a split request, until there are noMoreSplits().`